core/src/common.cpp

```diff
@@ -362,6 +364,434 @@ namespace irods::http
+			logging::error("{}: Error trying to parse jwks_uri [{}]. Please check configuration.", __func__, jwks_uri);
+			throw std::runtime_error{"Invalid [jwks_uri]."};
+		}
+
+		const auto url{*parsed_uri};
+		const auto port{get_port_from_url(url)};
+
+		// Addr
+		net::io_context io_ctx;
+		auto tcp_stream{irods::http::transport_factory(url.scheme_id(), io_ctx)};
+		tcp_stream->connect(url.host(), *port);
+
+		// Build Request
+		constexpr auto http_version_number{11};
+		beast::http::request<beast::http::string_body> req{beast::http::verb::get, url.path(), http_version_number};
+		req.set(beast::http::field::host, irods::http::create_host_field(url, *port));
+		req.set(beast::http::field::user_agent, irods::http::version::server_name);
+		req.set(beast::http::field::accept, "application/json");
+		req.prepare_payload();
+
+		// Send request & receive response
+		auto res{tcp_stream->communicate(req)};
+
+		logging::debug("{}: Received the following response: [{}]", __func__, res.body());
+
+		return res.body();
+	}
+
+	/// Adds the specified symmetric algorithm \p _alg to the verifier \p _verifier, using the secrets provided in the
+	/// HTTP API configuration
+	///
+	/// See RFC 7518 for details on JSON Web Algorithms (JWA)
+	///
+	/// \param[in,out] _verifier The jwt::verifier to add additional verification algorithms to.
+	/// \param[in]     _alg      The signing algorithm requested by the signed JWT.
+	auto add_symmetric_alg(
+		jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier,
+		std::string_view _alg) -> void
+	{
+		namespace logging = irods::http::log;
+
+		auto algorithm_family{_alg.substr(0, 2)};
+		auto algorithm_specifics{_alg.substr(2)};
+
+		// Extract the secret key
+		// OpenID secret should be our secret
+		std::string key;
+		if (auto realm_secret{irods::http::globals::oidc_configuration().find("realm_secret")};
+		    realm_secret != std::end(irods::http::globals::oidc_configuration()))
+		{
+			key = jwt::base::decode<jwt::alphabet::base64url>(jwt::base::pad<jwt::alphabet::base64url>(realm_secret->get<std::string>()));
+		}
+		else if (auto secret{irods::http::globals::oidc_configuration().find("client_secret")};
+		         secret != std::end(irods::http::globals::oidc_configuration()))
+		{
+			key = secret->get<std::string>();
+		}
+
+		if (algorithm_family == "HS") {
+			if (algorithm_specifics == "256") {
+				logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+				_verifier.allow_algorithm(jwt::algorithm::hs256(key));
+			}
+			else if (algorithm_specifics == "384") {
+				logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+				_verifier.allow_algorithm(jwt::algorithm::hs384(key));
+			}
+			else if (algorithm_specifics == "512") {
+				logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+				_verifier.allow_algorithm(jwt::algorithm::hs512(key));
+			}
+			return;
```

Comment:
What happens in the `else` case? It is neither added to the allowed algorithms, nor is a warning printed. It feels like it should drop through here or log a warning, rather than just `return`.